katello/katello-backup

```diff
@@ -63,59 +104,119 @@ def create_directories(directory)
   FileUtils.chmod_R 0770, directory
 end
 
-def backup_db_online
-  puts "Backing up postgres online schema... "
-  run_cmd("runuser - postgres -c 'pg_dumpall -g > #{@dir}/pg_globals.dump'")
-  run_cmd("runuser - postgres -c 'pg_dump -Fc foreman > #{@dir}/foreman.dump'")
-  run_cmd("runuser - postgres -c 'pg_dump -Fc candlepin > #{@dir}/candlepin.dump'")
-  puts "Done."
-
-  puts "Backing up mongo online schema... "
-  run_cmd('mongodump --host localhost --out mongo_dump')
-  puts "Done."
+def snapshot_backup
+  @mountdir = @options[:snapshot_mount] || "/var/snap"
+  @snapsize = @options[:snapshot_size] || "2G"
+  FileUtils.mkdir_p @mountdir
+  run_cmd("katello-service stop #{EXCLUDED}")
+  create_and_mount_snapshots
+  run_cmd("katello-service start #{EXCLUDED}")
+  backup_and_destroy_snapshots
 end
 
-def backup_pulp_online
-  puts "Backing up Pulp data... "
-  matching = false
- 
-  until matching
-    checksum1 = `find /var/lib/pulp -printf '%T@\n' | md5sum`
-    create_pulp_data_tar
-    checksum2 = `find /var/lib/pulp -printf '%T@\n' | md5sum`
-    matching = (checksum1 == checksum2)
+def create_and_mount_snapshots
+  @databases.each do |database|
+    puts "Creating and mounting #{database} snapshot"
+    lv_info = get_lv_info(database)
+    run_cmd("lvcreate -n#{database}-snap -L#{@snapsize} -s #{lv_info[0]}")
+    FileUtils.mkdir_p "#{File.join(@mountdir, database)}"
+    snapshot_location = run_cmd("lvs --noheadings -o lv_path -S lv_name=#{database}-snap").strip
+    options = lv_info[1] == 'xfs' ? "-onouuid,ro" : "-oro"
+    run_cmd("mount #{get_snapshot_location(database)} #{File.join(@mountdir, database)} #{options}")
```

Comment:
@evgeni - ah, i fixed it, that method didn't make it in to the script i pushed. Please try again, working now!